graphite/graphite_tcp.go

```diff
@@ -40,22 +44,45 @@ func (t *TCPServer) ListenAndServe(iface string) error {
 	if err != nil {
 		return err
 	}
+	t.listener = &ln
+
+	t.Logger.Println("listening on TCP connection", ln.Addr().String())
+	t.wg.Add(1)
 	go func() {
+		defer t.wg.Done()
 		for {
 			conn, err := ln.Accept()
 			if err != nil {
 				t.Logger.Println("error accepting TCP connection", err.Error())
-				continue
+				return
```

Comment:
I don't think we want to return here.  `Accept` could return temporary errors (like the other end aborted before we could accept, etc.), and the listener could still be intact.  If we return here, we are effectively ignoring all TCP traffic from the listener now.